tests/queries/0_stateless/02882_replicated_fetch_checksums_doesnt_match.sql

```diff
@@ -0,0 +1,42 @@
+DROP TABLE IF EXISTS r1;
+DROP TABLE IF EXISTS r2;
+DROP TABLE IF EXISTS r3;
+
+CREATE TABLE checksums_r1 (column1 UInt32, column2 String) Engine = ReplicatedMergeTree('/tables/{database}/checksums_table', 'r1') ORDER BY tuple();
+
+CREATE TABLE checksums_r2 (column1 UInt32, column2 String) Engine = ReplicatedMergeTree('/tables/{database}/checksums_table', 'r2') ORDER BY tuple();
+
+CREATE TABLE checksums_r3 (column1 UInt32, column2 String) Engine = ReplicatedMergeTree('/tables/{database}/checksums_table', 'r3') ORDER BY tuple();
+
+SYSTEM STOP REPLICATION QUEUES checksums_r2;
+SYSTEM STOP REPLICATION QUEUES checksums_r3;
+
+ALTER TABLE checksums_r1 MODIFY COLUMN column1 Int32 SETTINGS alter_sync=1;
+
+INSERT INTO checksums_r1 VALUES (1, 'hello');
+
+INSERT INTO checksums_r3 VALUES (1, 'hello');
+
+SYSTEM START REPLICATION QUEUES checksums_r2;
+
+SYSTEM SYNC REPLICA checksums_r2;
+
+SELECT count() FROM checksums_r1;
+SELECT count() FROM checksums_r2;
+SELECT count() FROM checksums_r3;
```

Comment:
So, it will be possible that we don't see data on a particular replica right after inserting to this replica, until part with the expected checksum is fetched, right?

Reply:
Right, and I think it's more expected behavior.